client/X11/xf_monitor.c

```diff
@@ -281,8 +282,15 @@ BOOL xf_detect_monitors(xfContext* xfc, UINT32* pMaxWidth, UINT32* pMaxHeight)
 				*pMaxHeight = (xfc->workArea.height * settings->PercentScreen) / 100;
 		}
 	}
+	else if (settings->DesktopWidth || settings->DesktopHeight)
+	{
+		if (settings->DesktopWidth)
+			*pMaxWidth = settings->DesktopWidth;
+		if (settings->DesktopHeight)
+			*pMaxHeight = settings->DesktopHeight;
+	}
 
-	if (!settings->Fullscreen && !settings->Workarea && !settings->UseMultimon)
+	if (!settings->Fullscreen && !settings->Workarea && !settings->UseMultimon && !(settings->DesktopWidth || settings->DesktopHeight))
```

Comment:
@T3N4K4 This is the culprit for your weired rectangle, the new checks are wrong.